pkgs/build-support/node/build-npm-package/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchNpmDeps, callPackage, nodejs }:
+
+{ name ? "${args.pname}-${args.version}"
+, src ? null
+, srcs ? null
+, sourceRoot ? null
+, patches ? [ ]
+, npmDepsHash ? ""
+, nativeBuildInputs ? [ ]
+, buildInputs ? [ ]
+, ...
+} @ args:
+
+assert npmDepsHash != "";
+
+let
+  npmDeps = fetchNpmDeps {
+    inherit src srcs sourceRoot patches;
```

Comment:
IIRC `srcOnly` has support for multiple `srcs`, can probably be used here. Then you don't need to account for patches or potential of multiple `srcs`.

Reply:
I just checked and it turns out it does. Should definitely switch to it so we don't need `patches`, `src` or `srcs`.

Current implementation also does not account for patches done dynamically in `postPatch` using `substituteInPlace`, which `srcOnly` fixes.

Reply:
After talking in private about this, Ivar realized she was confused about what this was for, so I'm resolving this.